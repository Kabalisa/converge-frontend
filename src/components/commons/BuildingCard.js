/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

import '../../assets/styles/officeStructure.scss';

const setClassName = (block, activeBlock) => {
  const className = `office-structure-block ${
    block === activeBlock
      ? 'office-structure-block active-block'
      : 'office-structure-block'
  }`;
  return className;
};
const BuildingCard = ({ blocks, activeBlock, toggleBlock }) => (
  <div className="office-structure-container">
    {blocks.map(block => (
      <button
        className={setClassName(block.name, activeBlock)}
        onClick={() => toggleBlock(block.name)}
      >
        {block.name}
      </button>
    ))}
  </div>
);

BuildingCard.propTypes = {
  blocks: PropTypes.array,
  activeBlock: PropTypes.string,
  toggleBlock: PropTypes.func.isRequired,
};

BuildingCard.defaultProps = {
  blocks: [],
  activeBlock: '',
};

export default BuildingCard;
