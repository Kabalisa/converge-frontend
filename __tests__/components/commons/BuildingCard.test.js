import React from 'react';
import { shallow } from 'enzyme';
import BuildingCard from '../../../src/components/commons/BuildingCard';

describe('BuildingCard component', () => {
  const props = {
    blocks: ['Block A', 'Block B'],
    activeBlock: 'Block A',
  };

  const wrapper = shallow(<BuildingCard {...props} />);

  it('should render with no errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
