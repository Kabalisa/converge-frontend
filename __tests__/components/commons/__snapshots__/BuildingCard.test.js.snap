// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingCard component should render with no errors 1`] = `
<div
  className="office-structure-container"
>
  <button
    className="office-structure-block office-structure-block"
    onClick={[Function]}
  />
  <button
    className="office-structure-block office-structure-block"
    onClick={[Function]}
  />
</div>
`;
